### ISSUE TITLE
How to reproduce the results when training custom dataset?

### ISSUE THREAD

**Issue by hugh-98** [ultralytics/yolov5#6237]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

When I trained my dataset, I foundthe mAP can't be reproduced!!! init_seeds(0) was set. And I used two GPUs.
My results that used the same command run are as followed:
![image](https://user-images.githubusercontent.com/33445477/148627648-34dc5405-c800-4e38-be92-3d7538945f54.png)

Does anyone encounter this situation???

**Comment by glenn-jocher** [comment]
@hugh-98 👋 Hello! Thanks for asking about **training reproducibility**. YOLOv5 🚀 uses a single training seed which is set here using the `init_seeds()` function.
https://github.com/ultralytics/yolov5/blob/6865d19a92d8c160c7fc3c92256627dadce1cd1e/train.py#L108

This function sets `python`, `numpy` and `torch` seeds and updates `cudnn` settings:
https://github.com/ultralytics/yolov5/blob/6865d19a92d8c160c7fc3c92256627dadce1cd1e/utils/general.py#L122-L130

To set a new seed for example:
```python
    init_seeds(2 + RANK)  # set seed=2
```

Note that even when using the same seed trainings may produce different results, especially when using CUDA backends. See https://pytorch.org/docs/stable/notes/randomness.html for details on factors affecting PyTorch training reproducibility and sources of randomness.

<img width="949" alt="Screen Shot 2022-01-08 at 11 20 59 AM" src="https://user-images.githubusercontent.com/26833433/148660366-9155dfd9-351c-4174-818f-2b1b907b3e4e.png">

Good luck 🍀 and let us know if you have any other questions!

**Comment by hugh-98** [comment]
@glenn-jocher I want to know that if I can't reproduce the result, can I use YOLOv5 as the baseline of paper?
I really want to use YOLOv5 as a baseline, then publish one thesis...😭😭😭
even if I use the method of https://pytorch.org/docs/stable/notes/randomness.html, I find I still can't reproduce my dataset.
It's too bad for me...
